Model/Cache.php

```diff
@@ -168,6 +172,29 @@ private function getContainer()
         return $this->container;
     }
 
+    /**
+     * Shore up EntityManager loading, in case there is a flaw in a plugin or campaign handling.
+     *
+     * @return EntityManager
+     */
+    private function getEntityManager()
```

Comment:
Maybe consider making this a trait? This is used frequently.. Could replace 'ContactClient' with get_class($this)